compiler/vm/vm.nim

```diff
@@ -1287,6 +1287,32 @@ proc rawExecute(c: var TCtx, pc: var int, tos: var StackFrameIndex): TFullReg =
               ast[i] = a.sym.ast[i]
             ast[bodyPos] = transformBody(c.graph, c.idgen, a.sym, cache=true)
             ast.copyTree()
+    of opcExpandToAst:
+      decodeBC(rkNode)
+
+      let callExprAst = c.constants[regs[rb].intVal.int]
+      let prc = callExprAst[0].sym
+
+      assert callExprAst.kind in nkCallKinds
+      assert prc.kind == skTemplate
+
+      let prevFrame = c.sframes[tos].next
+      let genSymOwner = if prevFrame > 0 and c.sframes[prevFrame].prc != nil:
+                          c.sframes[prevFrame].prc
+                        else:
+                          c.module
+      var templCall = newNodeI(nkCall, c.debug[pc])
+      templCall.add(newSymNode(prc))
+      for i in 1..rc-1:
+        let node = regs[rb+i].regToNode
+        node.info = c.debug[pc]
+        templCall.add(node)
+
+      var a = evalTemplate(templCall, prc, genSymOwner, c.config, c.cache, c.templInstCounter, c.idgen)
+      if a.kind == nkStmtList and a.len == 1: a = a[0]
```

Comment:
```suggestion
      if a.kind == nkStmtList and a.len == 1: # flatten if a single statement
        a = a[0]
```

Also, reading this code, what if it's empty? Should this be an nkEmpty then?

Reply:
I don't know. In order to preserve the previous behaviour, I left this as is

Reply:
OK, let's stick with it for now, it does feel like a bug waiting to happen. 😬